Lib/fontTools/subset/__init__.py

```diff
@@ -1635,7 +1635,7 @@ def prune_post_subset(self, font, options):
 	#	table.ScriptList = None
 
 	if hasattr(table, 'FeatureVariations'):
-		if not (table.FeatureList and table.FeatureVariations.FeatureVariationRecord):
+		if not (table.FeatureList and hasattr(table.FeatureVariations, 'FeatureVariationRecord') and table.FeatureVariations.FeatureVariationRecord):
```

Comment:
I think you should test for `table.FeatureVariations` being not `None`. I'm fairly sure that as soon as it's not `None` it will also have the `FeatureVariationRecord` attribute.